src/components/Admin/ProductItem.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from "react";
+
+export default class ProductItem extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { isEdit: false };
+    this.editProduct = this.editProduct.bind(this);
+    this.editProductSubmit = this.editProductSubmit.bind(this);
+    this.deleteProduct = this.deleteProduct.bind(this);
+  }
+  deleteProduct() {
+    const { id } = this.props.product;
+    this.props.deleteProduct(id);
+  }
+  editProduct() {
+    this.setState((prevState, props) => ({
+      isEdit: !prevState.isEdit,
+    }));
+  }
+  editProductSubmit() {
+    this.setState((prevState, props) => ({
+      isEdit: !prevState.isEdit,
+    }));
+    this.props.editProductSubmit(
+      this.props.product.id,
+      this.titleInput.value,
+      this.descriptionInput.value,
+      this.priceInput.value,
+      this.imageInput.value
+    );
+  }
+  render() {
+    const { title, description, price, image } = this.props.product;
+    return this.state.isEdit === true ? (
```

Comment:
why explicit comparing? i would prefer to use `const { isEdit } = this.state` => isEdit ? ... : ..